setup.cfg

```diff
@@ -1,5 +1,49 @@
-[aliases]
-test=pytest
+[metadata]
+name = PyMeasure
+# version is dynamically generated with setuptools_scm
+url = https://github.com/pymeasure/pymeasure
+author = PyMeasure Developers
+description = Scientific measurement library for instruments, experiments, and live-plotting
+long_description = file: README.rst, CHANGES.rst
+long_description_content_type = text/x-rst
+keywords = measure, instrument, experiment control, automate, graph, plot
+license = MIT
+license_files = LICENSE.txt
+classifiers =
+    Development Status :: 4 - Beta
+    Intended Audience :: Science/Research
+    License :: OSI Approved :: MIT License
+    Operating System :: MacOS
+    Operating System :: Microsoft :: Windows
+    Operating System :: POSIX
+    Operating System :: Unix
+    Programming Language :: Python :: 3 :: Only
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Topic :: Scientific/Engineering
+
+[options]
+packages = find:
+install_requires =
+    numpy >= 1.6.1
+    pandas >= 0.14
+    pyvisa >= 1.8
+    pyserial >= 2.7
+    pyqtgraph >= 0.9.10
+    importlib-metadata; python_version<"3.8"
+
+[options.extras_require]
+matplotlib = matplotlib>=2.0.2
+tcp =
```

Comment:
Is matplotlib needed by pymeasure ?

Reply:
Hm, I copied/moved over what was already there. That's maybe a leftover from earlier. Nowadays we require pandas, and probably assume matplotlib will be there (but it's not required by pandas).
Question to @CasperSchippers : is matplotlib mandatory? Does all our (experiment/GUI code) code work (meaning: could also fall back to something else if mpl is not found)? 
If optional, but needed to get plotting capability: should we rename that extra require target/feature selector `plotting` to be clearer?

Reply:
I think the only place where `matplotlib` is used is here https://github.com/pymeasure/pymeasure/blob/0400c37c9fa474afe8fbddb041de6f5bb708a259/pymeasure/display/curves.py#L36

This is optional and used to import the colormap

Reply:
...is used _directly_. What I did not have the time to check, yet, is what happens if we want to use pymeasure GUI plotting in an environment that does not have matplotlib installed.

Reply:
The complete code is as follow;
```
try:
    from matplotlib.cm import viridis
except ImportError:
    log.warning("Matplotlib not found. Images will be greyscale")
```
So when `matplotlib` is not available, GUI will still work and the `ResultsImage` will be grey scale.

In any case, I have submitted a PR #622 to clean all this and use colormap from `pyqtgraph`

Reply:
I just merged #622 and will remove the matplotlib extras_require entry.

Reply:
Huh, I double-checked, and there is [another use](https://github.com/pymeasure/pymeasure/blob/7369ff12cd969b744035839e443ee8894e1d3b56/pymeasure/experiment/config.py#L48) of matplotlib. 
However, this only gets hit if a user has defined an experiment config entry "matplotlib.rcParams", so presumably then they'll have matplotlib installed anyway. 

I think considering this, and the fact that the minimum version we require is very old by now, I think it is still safe to drop the explicit target.